### DISCUSSION TITLE
Flash writes, energy counter and brownout detection , can we do something ?

### DISCUSSION THREAD

**Discussion by ouinouin** [arendst/Tasmota#21226] [General]
Hi, i m using a lot of tamota energy monitoring devices home (about 20). As many users, i noticed that the energy counters (daily and total) got reset to the last value that was in flash on power outage conditions. 
I ask myself if there is a clever solution to that, since as i understand, there is a brownout detector on the microcontroller, can we use the brownout signal to ulimately write energy values to the flash so we dont get back to the last written value, that is basically the value upon the last reboot, (correct me if i m wrong, but tasmotA only writes those values upon a reboot?).
another clever solution would be to have an option to do  readback of the mqtt topic , and based on a brownout reboot to readback the value retained on mqtt.

id be glad to hear about electronicians/developpers  explaining what does the brownout allows ( how much time remains to do a task while trigger) , and yes surely depends on the capacitor upfront...

**Comment by barbudor** [comment]
Brownout detects a problem on the power supply
Once it triggered, it's too late. Esp is reset because memory is likely to be corrupted.
So too late to store anything

Only an hardware designed for could help which is not the case of those cheap devices

**Comment by sfromis** [comment]
My solution to that risk (not high where I live) is to use the telemetry data published via MQTT, which I save in my backend setup. That would enable me to detect a device coming online with too old data (totals lower than last reported), and publish commands to adjust. However, I don't really care much about the on-device totals, as I can already deal with such "blips" in my reporting setup.
